Emma/emma_libs/memoryManager.py

```diff
@@ -277,8 +283,13 @@ def drawElements(image, elementsToPlot, startPoint, y, colour):
                         plottedElements.append((element[Element.addressEnd] + len(element[Element.fqn]), currYLvl))
                     # FQN fits into element
                     else:
+                        # Check if address end of a drawing object is bigger than end point
+                        if element[Element.addressEnd] > endPoint:
+                            xAxeEnd = endPoint - startPoint - smallSpacing
+                        else:
+                            xAxeEnd = element[Element.addressEnd] - startPoint - smallSpacing
                         image.add(image.text(hex(originalStartAddress), insert=(xAxeRectStart + smallSpacing, currYLvl), font_size=str(fontSize)+"px", writing_mode="tb", font_family="Helvetica, sans-serif", fill=fontColour))
-                        image.add(image.text(hex(element[Element.addressEnd]), insert=(element[Element.addressEnd] - startPoint - smallSpacing, currYLvl), font_size='2px', writing_mode="tb", font_family="Helvetica, sans-serif", fill=fontColour))
+                        image.add(image.text(hex(element[Element.addressEnd]), insert=(xAxeEnd, currYLvl), font_size='2px', writing_mode="tb", font_family="Helvetica, sans-serif", fill=fontColour))
```

Comment:
Since `xAxeRectStart` does not include `smallspacing` it would be consistent to add it just here for `xAxeEnd` (not before).